### PR TITLE
ci: add soak testing as part of nightly build & test and remove Jenkinsfile

### LINKED ISSUE #654: test: move soak clusters from Jenkins to ADO
depends on https://github.com/Azure/aad-pod-identity/issues/586

### PR DESCRIPTION
**Pull request by chewong**
**Reason for Change**:

**Issue Fixed**:

fixes #654.

**Notes for Reviewers**:

### CODE REVIEW

**Comment by chewong** [comment]
Soak tests have passed: https://dev.azure.com/azure/aad-pod-identity/_build/results?buildId=12343&view=logs&jobId=011e1ec8-6569-5e69-4f06-baf193d1351e. Pointing the PR tests back to `.piplines/e2e-tests-template.yml`

File: test/e2e/e2e_suite_test.go [hunk 1]
```diff
@@ -101,3 +105,29 @@ func initScheme() *runtime.Scheme {
...
+	Eventually(func() (bool, error) {
+		if err := kubeClient.List(context.TODO(), pods, client.InNamespace(corev1.NamespaceDefault)); err != nil {
+			return false, err
+		}
+
+		return true, nil
+	}, framework.ListTimeout, framework.ListPolling).Should(BeTrue())
+
+	var micPods, nmiPods []corev1.Pod
+	for _, pod := range pods.Items {
```
Comment (aramase) [inline test/e2e/e2e_suite_test.go:120]: In general, I think more scalable option would be just do a filtered list with MIC and NMI pod labels

File: test/e2e/framework/pod/pod_helpers.go [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+
+	"github.com/Azure/aad-pod-identity/test/e2e/framework"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	corev1 "k8s.io/api/core/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// ListInput is the input of ListMICPodsI
```
Comment (aramase) [inline test/e2e/framework/pod/pod_helpers.go:15]: nit: s/ListMICPodsI/ListPods as this is used generically for listing pods

**Pull request merged**